lib/nostrum/struct/channel.ex

```diff
@@ -1,27 +1,125 @@
+  "<#766435015768539156>"
+  ```
+
+  ## Channel Cache
 
-  channel = %Nostrum.Struct.Channel{id: 280085880452939778}
-  Nostrum.Api.create_message!(280085880452939778, "#{Nostrum.Struct.Channel.mention(channel)}")
-  %Nostrum.Struct.Message{content: "<#280085880452939778>"}
+  The [`ChannelCache`](`Nostrum.Cache.ChannelCache`) module provides functionality for you to retreive information about any channel that your application can see. It provides two functions: [`get()/1`](`Nostrum.Cache.ChannelCache.get()/1`) and [`get!()/1`](`Nostrum.Cache.ChannelCache.get!()/1`).
+
+  ```elixir
+  Nostrum.Cache.ChannelCache.get!(827333533688397865)
+      %Nostrum.Struct.Channel{
+            application_id: nil,
+            bitrate: nil,
+            guild_id: 766435015768539156,
+            icon: nil,
+            id: 827333533688397865,
+            last_message_id: nil,
+            last_pin_timestamp: nil,
+            name: "announcements",
+            nsfw: false,
+            owner_id: nil,
+            parent_id: nil,
+            permission_overwrites: [
+              %Nostrum.Struct.Overwrite{
+                allow: 0,
+                deny: 2048,
+                id: 766435015768539156,
+                type: :role
+              }
+            ],
+            position: 1,
+            recipients: nil,
+            topic: nil,
+            type: 5,
+            user_limit: nil
+          }
   ```
+
+  More details of the cache can be found at `Nostrum.Cache.ChannelCache`.
+
+  ## Helper Functions
+
+  This module contains two functions for assisting with channel structs. `mention/1` to convert the channel into a mention as a string, and `link/1` to convert the channel into a hyperlink as a string. Further details and examples are detailed in the [Functions section of this module.](#functions)
+
+  ## Api Functions
+
+  The Nostrum Api contains numerous functions related to channels. Notable functions relating to channels are shown below.
+
+  - `Nostrum.Api.create_guild_channel/2`
+  - `Nostrum.Api.get_channel/1`
+  - `Nostrum.Api.modify_channel/3`
+  - `Nostrum.Api.delete_channel/2`
+  - `Nostrum.Api.add_pinned_channel_message/2`
+  - `Nostrum.Api.create_channel_invite/3`
+  - `Nostrum.Api.get_guild_channels/1`
+  - `Nostrum.Api.modify_guild_channel_positions/2`
+
+  > Note: This is not an exhaustive list, for full details please see the `Nostrum.Api` module documentation.
+
+  ## Channel Types
+  Channels take the shape of various types depending on their use and not all fields are always used. The currently implemented channel types are detailed below. The type of channel is determined by the `:type` field.
+
+  This diagram represents the regular channel types `0`, `2`, `5` and `13`.
+
+    ![Discord Channels](http://puu.sh/I4uW4/21e27a3385.png)
+
+    The currently implemented channel types are:
+
+  |     | Channel Type        |                                                                                       |
```

Comment:
Do you think it would make sense to separate out the discord names (like `GUILD_TEXT`) and the description of the channels here to make it easier to read?